Tools/clinic/clinic.py

```diff
@@ -960,7 +960,7 @@ def parser_body(
                 return_error = ('return NULL;' if simple_return
                                 else 'goto exit;')
                 parser_code = [libclinic.normalize_snippet("""
-                    if (nargs) {{
+                    if (nargs || kwnames) {{
```

Comment:
Can it be an empty tuple? You perhaps cannot trigger this in Python, but maybe using the C API?

Look at `_PyArg_NoKwnames()`.

Reply:
> Can it be an empty tuple? You perhaps cannot trigger this in Python, but maybe using the C API?

Good question! Let's find out.

Reply:
I'm unable to force it using `PyObject_Call` and `PyObject_Vectorcall`. Did you have another C API in mind?

Reply:
I do not know. Do you get here the same tuple as passed to `PyObject_Vectorcall()` if it is not empty?

Reply:
If I pass an empty tuple to `PyObject_Vectorcall`, I am actually able to trigger the error; the generated clinic code will get the same pointer.

I guess we should guard against such API misuse.

Reply:
```suggestion
                    if (nargs || (kwnames && PyTuple_GET_SIZE(kwnames))) {
```

Reply:
This change LGTM.